### PR TITLE
2704 - Material UI - Menu

### PR DESCRIPTION
**Pull request by LunAndrey**

### CODE REVIEW

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        menuFrame.paddingMenuList.is().displayed();
+        menuFrame.menuListFirstButton.click();
+        menuFrame.menu.is().displayed();
+    }
+
+    @Test
+    public void contextMenuTest() {
+        contextMenuPage.open();
+
+        menuFrame.contextMenu.is().displayed();
```
Comment (evasileva-epam) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java:67]: Please check menu that opens by mouse right click

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/navigation/Menu.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public void scrollToElement() {
+        core().jsExecute("scrollIntoView()");
+    }
+
+    @Override
+    public MenuAssert is() {
+        return new MenuAssert().set(this);
+    }
+
+    @Override
```
Comment (evasileva-epam) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/navigation/Menu.java:24]: Please add @JDIAction

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Test
+    public void selectedVerticalPositioningTest() {
+        selectedVerticalPositioningPage.open();
+
+        menuFrame.menu.click();
+        menuFrame.scrollMenuList.is().displayed();
+        menuFrame.menuListFirstButton.click();
+        menuFrame.menu.is().displayed();
```
Comment (evasileva-epam) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java:40]: Please check that selected value is seen near menu

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Test
+    public void selectedNoVerticalPositioningTest() {
+        selectedNoVerticalPositioningPage.open();
+
+        menuFrame.menu.click();
+        menuFrame.scrollMenuList.is().displayed();
+        menuFrame.menuListFirstButton.click();
+        menuFrame.menu.is().displayed();
```
Comment (evasileva-epam) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java:50]: Please check that selected value is seen near menu

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Test
+    public void menuWithIconsTest() {
+        menuWithIconsPage.open();
+
+        menuFrame.menu.click();
+        menuFrame.paddingMenuList.is().displayed();
+        menuFrame.menuIcon.is().displayedSvg();
+        menuFrame.menuListFirstButton.click();
+        menuFrame.menu.is().displayed();
```
[the remark this answers is not in the document]
Comment (LunAndrey) [reply jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java:78]: it checks at line 76 with displayedSvg method

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/navigation/MenuAssert.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+    @JDIAction("Assert that {name}'s child svg is visible")
+    public MenuAssert displayedSvg() {
+        try {
+            boolean isDisplayed = new Timer(base().getTimeout() * 1000L)
+                    .wait(() -> element().find(".MuiListItemIcon-root").isDisplayed());
+            jdiAssert(isDisplayed, Matchers.is(true));
+        } catch (AssertionError e){
+            new AssertionError("Svg not found");
+        }
```
[the remark this answers is not in the document]
Comment (LunAndrey) [reply jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/navigation/MenuAssert.java:31]: without try-catch block this method fails

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (evasileva-epam) [reply jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/navigation/MenuTests.java:95]: Do you mean line 92?I think it's good idea to check icons in opened menu, e.g. for first button

**Pull request merged**